src/jit/morph.cpp

```diff
@@ -6475,6 +6475,11 @@ GenTreePtr Compiler::fgMorphField(GenTreePtr tree, MorphAddrContext* mac)
                     addr->gtIntCon.gtFieldSeq = fieldSeq;
 
                     tree->SetOper(GT_IND);
+                    // The GTF_FLD_NULLCHECK is the same bit as GTF_IND_ARR_LEN.
+                    // We must clear it when we transform the node.
+                    // TODO-Cleanup: It appears that the GTF_IND_ARR_LEN flag is never checked, and note
```

Comment:
> GTF_IND_ARR_LEN flag [](start = 56, length = 21)

I guess you wanted to mean GTF_FLD_NULLCHECK flag is never checked here.

(From source search I see fgValuenumberTree() is checking for GTF_IND_ARR_LEN flag.